[12.0] [MIG] beesdoo_easy_my_coop

StCyr opened the pull request:
**Do not try merging yet.**

Following questions/problems still need to be answered/solved:

1. Migration 11.0->12.0: doc says `All <label> elements in views must have a for="" attribute.`. I'm wondering if a text is mandatory between the quotes in `for=""` (concerns file `views/subscription_templates.xml`)
2. Module won't load. I think it's because it depends on module `beesdoo_shift` which is still in version 9.0 in branch `12.0-mig-obeesdoo`

tfrancoi committed:
[TASK 0044_b] Gestion des shifts : visualiser le créneau sur la fiche du membre régulier + [ADD] Tab worker information for all information linked with the work in the cooperative + [ADD] share type for beescoop + [ADD] Specific module that make the link between beesdoo and easy_my_coop : compute the share_type base on the share bought

tfrancoi committed:
[FIX] use existing field for cooperator type + [CLEAN] hide useless tab for physical person

tfrancoi committed:
[FIX] Make the runbot green again, remove the warning

tfrancoi committed:
[IMP] 008-z-D : mettre à jour les filtres Cooperateur de base pour prendre coop effectif: Change the domain to take into account the field member

tfrancoi committed:
[IMP] 008-z-C : enlever le type de coopérateur sur anciens coopérateurs: when share number is 0, you have no cooperator type anymore

tfrancoi committed:
[FIX] Double header on partner form

tfrancoi committed:
[ADD] Can shop field on res.partner that take into account the cooperator type and the cooperator status

remytms committed:
[ADD] b_easy_my_coop: Info Session in new cooperator form

This adds, in the new cooperator form in the website, a checkbox that
the user can check if he already has taken part to an information
session.

This information will be saved in the partner object and will be used to
automatically check the info_session field of the cooperator_status when
it will be created by the wizard.

remytms committed:
[FIX] b_easy_my_coop: Info Session must not be required field

In the form to become a new cooperator, the field info_session_confirmed
was required but not in the form validation. This commit fixes that
incoherence but changing the form to let field info_session_confirmed
not required.

remytms committed:
[IMP] b_emc: Info session confirmation in new coop form

This lets user configure if the info session confirmation should be
shown and/or required in the new cooperator form.

remytms committed:
[FIX] b_emc: Function misspelled

remytms committed:
[FIX] b_emc: Info session not always displayed

The info session was not always displayed on the form to become new
cooperator, despite a good configuration.

Commit:
[FIX] Mandatory field not displayed

Mandatory configuration approval field must be displayed now.

Same modifications as in f406155d5c7250b773a70ec7b8b4d6dd6bdd3da7

StCyr committed:
[MIG] beesdoo_easy_my_coop: Migration to 10.0

StCyr committed:
[MIG] beesdoo_easy_my_coop: Migration to 11.0

StCyr committed:
[MIG] beesdoo_easy_my_coop: Migration to 12.0

houssine78 commented on beesdoo_easy_my_coop/views/partner.xml:
```diff
@@ -0,0 +1,47 @@
+            <xpath expr="header" position="replace" />
+            <xpath expr="//field[@name='state']" position="before">
+                <button string="Create Subscription" name="%(easy_my_coop.action_view_create_subscription)d"
+                                type="action" context="{'default_active_id': active_id}" groups="easy_my_coop.group_energiris_manager"/>
```
groups name changed in 12. see https://github.com/coopiteasy/vertical-cooperative/blob/12.0/easy_my_coop/security/res_groups.xml here.

houssine78 commented on beesdoo_easy_my_coop/views/partner.xml:
```diff
@@ -0,0 +1,47 @@
+                <button string="Create Subscription" name="%(easy_my_coop.action_view_create_subscription)d"
+                                type="action" context="{'default_active_id': active_id}" groups="easy_my_coop.group_energiris_manager"/>
+                 <button string="Update info" name="%(easy_my_coop.action_view_update_partner_info)d"
+                                type="action" context="{'default_active_id': active_id}" groups="easy_my_coop.group_energiris_manager"/>
```
same here

houssine78 commented on beesdoo_easy_my_coop/views/partner.xml:
```diff
@@ -0,0 +1,47 @@
+            </filter>
+            <filter name="type_c" position="replace">
+                <filter name="type_c" string="Cooperator type C" domain="[('cooperator_type', '=', 'share_c'), ('member', '=', True)]" />
+            </filter>
```
it's no more hard coded an it's not generic. so this search view can be removed.

houssine78 commented on beesdoo_easy_my_coop/views/partner.xml:
```diff
@@ -0,0 +1,47 @@
+		<field name="inherit_id" ref="beesdoo_base.beesdoo_partner_form_view" />
+		<field name="arch" type="xml">
+			<field name="cooperator_type" position="replace" />
+		</field>
```
I would rather see if it still a duplicate in the view. Anyway it's no more a problem on odoo 12.

houssine78 commented on beesdoo_easy_my_coop/models/res_partner.py:
```diff
@@ -0,0 +1,44 @@
+    def _auto_init(self, cr, context=None):
+        cr.execute("ALTER TABLE res_partner DROP COLUMN IF EXISTS cooperator_type")
+        res = super(Partner, self)._auto_init(cr, context=context)
+        return res
```
can be removed as code is pretty similar now in easy my coop

houssine78 commented on beesdoo_easy_my_coop/models/res_partner.py:
```diff
@@ -0,0 +1,44 @@
+class Partner(models.Model):
+    _inherit = 'res.partner'
+
+    cooperator_type = fields.Selection(selection='_get_share_type', compute='_compute_share_type', string='Cooperator Type', store=True)
```
remove field override

houssine78 commented on beesdoo_easy_my_coop/models/res_partner.py:
```diff
@@ -0,0 +1,44 @@
+    @api.depends('cooperator_type', 'cooperative_status_ids', 'cooperative_status_ids.can_shop')
+    def _can_shop(self):
+        for rec in self:
+            if rec.cooperator_type == 'share_b':
```
this should not be hardcoded anymore. a boolean can_shop on the share(product.template) should do the work

remytms replied:
There is already a flag in easy_my_coop share product named `customer` that do the same thing then `can_shop`. So I've removed `can_shop` on the share product.

See : https://github.com/coopiteasy/vertical-cooperative/blob/12.0/easy_my_coop/models/product.py#L21

houssine78 replied:
as far that the can_shop on res_partner is handled and correctly mapped to customer field from the product, it is fine for me.

houssine78 reviewed (changes requested):
some adaptation should be brough. Thanks for the work so far !

houssine78 committed:
[IMP] add can shop boolean on the share product

It will avoid to test on hardcoded values. It also makes several shares
allowing to shop product from the coop

houssine78 committed:
[REFACT] refactor the code make can shop extendable and configurable.

houssine78 committed:
[IMP] remove unnecessary views and update security groups

houssine78 committed:
[FIX] fix module name in import

houssine78 committed:
[FIX] fix missing underscore for _inherit

vvrossem committed:
[FIX] beesdoo_easy_my_coop: correct XML external ID

vvrossem committed:
[FIX] beesdoo_easy_my_coop: missing dependency

remytms committed:
[REF] b_base: @api.one -> self.ensure_one()

remytms committed:
[CHG] b_base: country_id is not belgium by default anymore

remytms committed:
[CHG] b_base: Remove cooperator_type info

This new b_base module focus on adding member cards and eater
functionality.

Constrain about number of eater should be customise for each
cooperative. And if the cooperative is using easy_my_coop module the
maximum number of eater should be specified on the share product.

Also cooperator_type has no meaning here. For cooperative that uses
easy_my_coop, cooperator_type is a computed field based on the share
that the cooperator owns.

beesdoo_easy_my_coop and beesdoo_worker_status will be modified for
taking this into account.

remytms committed:
[CHG] b_emc: Eater max is configurable

remytms committed:
[FIX] b_worker_status: is_worker computation in b_emc

remytms committed:
[MOV] b_shift: res.partner definition in separated file

remytms committed:
[IMP] b_emc: Merge emc_eater into b_emc

remytms committed:
[FIX] b_base, b_emc: writing several eater at once

remytms committed:
[ADD] b_shift,b_emc: can_shop status on res_partner

remytms committed:
[FIX] b_emc: Constraint when editing parent_eater_id

remytms committed:
[FIX] b_shift,b_base: can_shop computation on res_partner
